sklearn/naive_bayes.py

```diff
@@ -1168,8 +1167,7 @@ def _check_X_y(self, X, y):
         #                    force_all_finite=True)
         X, y = check_X_y(X, y, accept_sparse=False, force_all_finite=True)
         X, y = check_X_y(X, y, dtype='int')
-        if np.any(X < 0):
-            raise ValueError("X must not contain negative values.")
+        check_non_negative(X, "CategoricalNB (input X)")
```

Comment:
By the way, `check_non_negative` finds the min and checks if it's negative, wouldn't it be faster to use `np.any` ? (in theory it would just need to iterate until finding one nonnegative element but I don't know if that's what is done by any)

Reply:
%timeit is your friend

Reply:
That's right, with timeit `np.min()` is faster. I forgot but in `np.any` we still need to do the `X>0` pass, which amounts to one pass on the whole data that's why it'll not be faster... I didn't find any numpy function that would prevent to do this whole pass so I guess `np.min` is good

```python
In[1]: import numpy as np

In [2]: a = np.random.randn(10000, 10000)              
                                                                                    
In [3]: %timeit np.any(a)                                                                                                                  
124 ms ± 808 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [4]: %timeit np.min(a)                                                                                                                  
56.5 ms ± 69.8 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```